source/Color.cpp

```diff
@@ -130,3 +136,67 @@ Color Color::Multiply(float scalar, const Color &base)
 			scalar * base.color[2],
 			scalar * base.color[3]);
 }
+
+
+
+// Apply color blindness filters to this color.
+Color Color::Filter(Color c)
+{
+	Preferences::ColorFilter filter = Preferences::GetColorFilterMode();
+	// No color filters are enabled, so there's no need to adjust anything.
+	if(filter == Preferences::ColorFilter::NORMAL)
+		return c;
+
+	// Color blindness accessibility filters are enabled.
+	// LMS Daltonization is used to make the colors more distinct for
+	// color-blind players.
+	static float l, m, s, r, g, b;
+
+	// Convert the colors from RGB to LMS, a color space that represents the
+	// light received by the cones in the human eye better than RGB.
+	l = (17.8824 * c.color[0]) + (43.5161 * c.color[1]) + (4.11935 * c.color[2]);
+	m = (3.45565 * c.color[0]) + (27.1554 * c.color[1]) + (3.86714 * c.color[2]);
+	s = (0.0299566 * c.color[0]) + (0.184309 * c.color[1]) + (1.46709 * c.color[2]);
+
+	switch(filter)
+	{
+		// Simulate color blidness.
+		case Preferences::ColorFilter::PROTANOPIA:
+			l = (2.02344 * m) + (-2.52581 * s);
+			break;
+		case Preferences::ColorFilter::DEUTERANOPIA:
+			m = (0.494207 * l) + (1.24827 * s);
+			break;
+		case Preferences::ColorFilter::TRITANOPIA:
+			s = (-0.395913 * l) + (0.801109 * m);
+			break;
+		default:
+			break;
```

Comment:
Not necessary to have this; can be removed.

Reply:
Without this, CMake throws a whole bunch of errors.

Reply:
oh yeah, change it to the NORMAL value instead. mb.